packages/nextjs/src/performance/client.ts

```diff
@@ -1,13 +1,104 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 
-import { Primitive, Transaction, TransactionContext } from '@sentry/types';
-import { fill, getGlobalObject, stripUrlQueryAndFragment } from '@sentry/utils';
+import { getCurrentHub } from '@sentry/hub';
+import { Primitive, TraceparentData, Transaction, TransactionContext } from '@sentry/types';
+import {
+  extractTraceparentData,
+  fill,
+  getGlobalObject,
+  logger,
+  parseBaggageHeader,
+  stripUrlQueryAndFragment,
+} from '@sentry/utils';
+import type { NEXT_DATA as NextData } from 'next/dist/next-server/lib/utils';
 import { default as Router } from 'next/router';
+import type { ParsedUrlQuery } from 'querystring';
 
 const global = getGlobalObject<Window>();
 
 type StartTransactionCb = (context: TransactionContext) => Transaction | undefined;
 
+/**
+ * Describes data located in the __NEXT_DATA__ script tag. This tag is present on every page of a Next.js app.
+ */
+interface SentryEnhancedNextData extends NextData {
+  // contains props returned by `getInitialProps` - except for `pageProps`, these are the props that got returned by `getServerSideProps` or `getStaticProps`
+  props: {
+    _sentryGetInitialPropsTraceData?: string; // trace parent info, if injected by server-side `getInitialProps`
+    _sentryGetInitialPropsBaggage?: string; // baggage, if injected by server-side `getInitialProps`
+    pageProps?: {
+      _sentryGetServerSidePropsTraceData?: string; // trace parent info, if injected by server-side `getServerSideProps`
+      _sentryGetServerSidePropsBaggage?: string; // baggage, if injected by server-side `getServerSideProps`
```

Comment:
I know we're leaving `getStaticProps` somewhat to the side for the moment, but do you think it's worth it to add a TODO about it here?

Reply:
There sadly is no way to return trace data from `getStaticProps`. I finally properly read up on it. When a page needs to be revalidated (so that `getStaticProps` is run during run-time), the user receives a fallback page first, which doesn't yet have any data from `getStaticProps`. When `getStaticProps` is done fetching it will send the data over. At no point in time the data is in `__NEXT_DATA__`.

I generally feel like properly tracing `getStaticProps` is gonna be hard/impossible without `instrumentServer` but let's have that discussion when we get to it.

Reply:
See https://github.com/getsentry/sentry-javascript/pull/5574#discussion_r946537752